commands.go

```diff
@@ -1,8 +1,24 @@
 package raft
 
+// VersionInfo is a common sub-structure used to pass along
+// protocol version information. For older Raft implementations before
+// versioning was added this will default to protocol version 0.
+type VersionInfo struct {
```

Comment:
That's a good idea and this is exactly where that would go. How about `RPCHeader`?